backend/app/Http/Controllers/RoomController.php

```diff
@@ -95,4 +93,23 @@ public function destroy(Request $request, Room $room): JsonResponse
         $room->delete();
         return response()->json(null, 204);
     }
+
+
+    /**
+     * Generate a random temporary room code for this room. 
+     */
+    public function generateRoomCode(Room $room): JsonResponse
+    {
```

Comment:
please authenticate here, check if the user is allowed to access the room (auth:sanctum only assures that any user is logged in, not the user that owns this room)